client/chaincode/system/lifecycle.go

```diff
@@ -63,6 +66,66 @@ func (c *lifecycleCC) InstallChaincode(ctx context.Context, installArgs *lb.Inst
 	return ccResult, nil
 }
 
+// ApproveFromMyOrg approves chaincode package on a channel
+func (c *lifecycleCC) ApproveFromMyOrg(
+	ctx context.Context,
+	channelID string,
+	broadcastClient api.Orderer,
+	approveArgs *lb.ApproveChaincodeDefinitionForMyOrgArgs,
+) error {
+	var (
+		args      []byte
+		resp      *peer.ProposalResponse
+		processor api.PeerProcessor
+		tx        api.ChaincodeTx
+		err       error
+	)
+
+	if args, err = proto.Marshal(approveArgs); err != nil {
+		return fmt.Errorf("marshal args: %w", err)
+	}
+	processor = peerSDK.NewProcessor(channelID)
+
+	prop, tx, err := processor.CreateProposal(
+		lifecycleName,
+		c.identity,
+		lifecycle.ApproveChaincodeDefinitionForMyOrgFuncName,
+		[][]byte{args},
+		nil,
+	)
+	if err != nil {
+		return errors.Wrap(err, `failed to create proposal`)
```

Comment:
we use fmt.Errorf for new and modified code

Reply:
fixed